src/Models/IoTContainerizedProject.ts

```diff
@@ -147,10 +147,21 @@ export class IoTContainerizedProject extends IoTWorkbenchProjectBase {
           `Internal Error. Could not find iot workbench project file.`);
     }
 
-    // Open project
     await this.openProject(this.projectRootPath, openInNewWindow);
   }
 
+  async openFolderInContainer(folderPath: string) {
+    if (!await FileUtility.directoryExists(ScaffoldType.Local, folderPath)) {
+      throw new Error(
+          `Fail to open folder in container. ${folderPath} does not exist.`);
+    }
+
+    setTimeout(

```

Comment:
why need setTimeout here?

Reply:
Wait for the telemetry to be sent before VS Code restart.

Reply:
Better to use await telemetry send then trigger this command?

Reply:
Currently TelemetryRepoter does not provide async `sendTelemetryEvent` function. I have opened an issue on their repo.

Reply:
Just check the repo. One question here, is telemetry.sendTelemetyrEvent really an async method? Or it is not async, since the message just need to be sendout and no need to wait for the response?

Reply:
Verified and remote the `setTimeout` machenism.